migrations/20231129172339_job_queue_table.sql

```diff
@@ -0,0 +1,16 @@
+CREATE TYPE job_status AS ENUM(
+    'available',
+    'completed',
+    'failed',
+    'running'
+);
+
+CREATE TABLE job_queue(
+    id BIGSERIAL PRIMARY KEY,
+    attempt INT NOT NULL DEFAULT 0,
+    created_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+    finished_at TIMESTAMPTZ DEFAULT NULL,
+    started_at TIMESTAMPTZ DEFAULT NULL,
+    status job_status NOT NULL DEFAULT 'available'::job_status,
+    parameters JSONB
```

Comment:
It was discussed. The design of the system includes a global management service that will disable bad endpoints by querying PG. This service will disable _all_ webhooks of the same endpoint. It will be separate from the queue system, but we may need to extend the schema to support this. Perhaps with an endpoint column?

Reply:
For example: river queue provides a `kind` field that uniquely identifies a type of job. In our context, this could be `endpoint` and uniquely identify a webhook endpoint. Continuing with river queue as an example, they also have a `"discarded"` state to identify jobs that have failed for the last time.

Reply:
Cool, that makes sense. And yeah, it does seem like we'll want an easier column to query than the raw JSONB. Could be added in later iterations though, up to you.

Reply:
I'll open a follow-up PR to discuss retries, just to keep things separate.

Reply:
I didn't open a follow-up PR and instead implemented it here...